src/frontend/AppContainerWrapper.tsx

```diff
@@ -108,45 +136,19 @@ const AppContainerWrapper = () => {
     []
   );
 
-  const onNavStateChange = React.useCallback(
-    (previousState: NavigationState, currentState: NavigationState) => {
-      const previousRouteName = getRouteName(previousState);
-      const currentRouteName = getRouteName(currentState);
-
-      if (previousRouteName !== currentRouteName) {
-        updateRouteBasedAppState(currentRouteName);
-      }
-    },
-    [updateRouteBasedAppState]
-  );
-
   const openInviteModal = React.useCallback((key: string) => {
-    if (navRef.current) {
-      navRef.current.dispatch(
-        NavigationActions.navigate({
-          routeName: "ProjectInviteModal",
-          params: { invite: key },
-        })
-      );
+    if (!!navRef) {
+      navRef.navigate("ProjectInviteModal", { inviteKey: key });
     }
   }, []);
 
-  const { loadNavigationState, persistNavigationState } = React.useMemo(
-    () =>
-      IS_E2E
-        ? {}
-        : {
-            loadNavigationState: async () => {
-              const loadedNavState = await loadSavedNavState();
-
-              const loadedRouteName = getRouteName(loadedNavState);
-
-              updateRouteBasedAppState(loadedRouteName);
-
-              return loadedNavState;
-            },
-            persistNavigationState: createNavigationStatePersister(),
-          },
+  const handleNavStateChange = React.useCallback(
+    (navState?: NavigationState) => {
+      if (!navState) return;
+      //TO DO: Check if prev route !== current route
+      updateRouteBasedAppState(getRouteName(navState));
+      createNavigationStatePersister(navState);
```

Comment:
Why do we want to create this (navigationStatePersister) every time this callback runs?